chef/cookbooks/repos/recipes/default.rb

```diff
@@ -57,10 +59,10 @@
     end
     package "rubygems"
   when "redhat","centos"
-    maj,min = node[:platform_version].split(".",2)
-    repositories = Range.new(0,min.to_i).to_a.reverse.map{|v|
-      provisioner["provisioner"]["repositories"]["#{node[:platform]}-#{maj}.#{v}"] rescue nil
-    }.compact.first
+    maj, min = node[:platform_version].split(".", 2)
+    repositories = Range.new(0, min.to_i).to_a.reverse.map do |v|
+      provisioner["provisioner"]["repositories"]["#{node[:platform]}-#{maj}.#{v}"][arch] rescue nil
```

Comment:
Style/RescueModifier: Avoid using `rescue` in its modifier form. (https://github.com/bbatsov/ruby-style-guide#no-rescue-modifiers)